Code/Legacy/CrySystem/DebugCallStack.cpp

```diff
@@ -896,13 +778,40 @@ static bool IsFloatingPointException(EXCEPTION_POINTERS* pex)
     }
 }
 
-int DebugCallStack::PrintException(EXCEPTION_POINTERS* exception_pointer)
+DebugCallStack::UserPostExceptionChoice DebugCallStack::AskUserToRecoverOrCrash(EXCEPTION_POINTERS* exception_pointer)
 {
-    return (int)DialogBoxParam(gDLLHandle, MAKEINTRESOURCE(IDD_CRITICAL_ERROR), NULL, DebugCallStack::ExceptionDialogProc, (LPARAM)exception_pointer);
+    if (exception_pointer->ExceptionRecord->ExceptionFlags & EXCEPTION_NONCONTINUABLE)
+        return UserPostExceptionChoice::Exit;
+
+    UserPostExceptionChoice res = UserPostExceptionChoice::Exit;
+    if (auto nativeUI = AZ::Interface<AZ::NativeUI::NativeUIRequests>::Get(); nativeUI != nullptr)
+    {
+        DebugCallStack* pDCS = static_cast<DebugCallStack*>(DebugCallStack::instance());
+        AZStd::string msg = AZStd::string::format(
+            "O3DE encountered an error but can recover from it.\nDo you want to try to recover ?\n\nException Code: %s\nException Addr: "
+            "%s\nException Module: %s\nException Description: %s\nCallstack:\n%.600s",
+            pDCS->m_excCode,
+            pDCS->m_excAddr,
+            pDCS->m_excModule,
+            pDCS->m_excDesc,
+            pDCS->m_excCallstack);
+
+        constexpr bool showCancel = false;
+        AZStd::string dialogRes = nativeUI->DisplayYesNoDialog("Try to recover ?", msg, showCancel);
+        if (dialogRes == "Yes")
+            res = UserPostExceptionChoice::Recover;
```

Comment:
missing curly braces (see above)
```suggestion
        if (dialogRes == "Yes")
        {
            res = UserPostExceptionChoice::Recover;
        }
```